app/models/health/qualifying_activity.rb

```diff
@@ -551,37 +552,50 @@ def maintain_procedure_valid
     end
 
     def compute_valid_unpayable?
```

Comment:
The should probably reference the computed value if it's set, maybe?  Seems like we'd like to minimize the number of times the computation is run.